code/modules/awaymissions/mission_code/ruins/deepstorage.dm

```diff
@@ -282,6 +282,53 @@
 	name = "portal"
 	desc = "Good luck."
 
+/obj/effect/abstract/deepstorage_cheese_detector
+	name = "deepstorage anti-theft system"
+
+/obj/effect/abstract/deepstorage_cheese_detector/Crossed(atom/movable/AM, oldloc)
+	. = ..()
+	var/msg = "An unauthorized organism has been detected within the perimeter. Initiating cleaning process."
+	if(ishuman(AM) || isrobot(AM))
+		if(length(GLOB.fleshling) != 0)
+			for(var/mob/R in range(35, src))
+				to_chat(R, "<b><font size=4 color=red>Intruder Alert</font><br> <font size=3><span class='robot'>[msg]</font size></font></b></span>")
+				SEND_SOUND(R, sound('sound/misc/notice1.ogg'))
+			explosion(loc, 3, 5, 6, flame_range = 6)
+		for(var/obj/effect/abstract/deepstorage_cheese_detector/DCD in urange(10, get_turf(src)))
+			qdel(DCD)
+
+/obj/item/documents/syndicate/unknown_specimen
```

Comment:
this item path would allow them to be used as the document theft objective.

Reply:
oh right, this isn't good. I'll probably have to make a change in supply.dm and change the path but that's for another day, supply.dm gives me headache